specification/dsdl/serialization.tex

```diff
@@ -568,20 +562,166 @@ \subsubsection{Tagged unions}
 
     \begin{minted}{python}
         # Empty. The only valid serialized representation is an empty bit sequence.
+        @final
     \end{minted}
 
-    Consider the following union schema definition:
+    Consider the following union:
 
     \begin{minted}{python}
+        @final
         @union
         Empty.1.0 none
         AnyType.1.0 some
     \end{minted}
 
-    The set of serialized representations of the union schema definition given above is equivalent to
+    The set of serialized representations of the union given above is equivalent to
     that of the following variable-length array:
 
     \begin{minted}{python}
+        @final
         AnyType.1.0[<=1] maybe_some
     \end{minted}
 \end{remark}
+
+\subsubsection{Delimited types}\label{sec:dsdl_serialization_composite_non_final}
+
+Objects of delimited (non-final) composite types that are nested inside other objects\footnote{%
+    Of any type, not necessarily composite; e.g., arrays.
+}
+are serialized into opaque containers that consist of two parts:
+the fixed-length \emph{delimiter header},
+immediately followed by the serialized representation of the object as if it was of a final type.
+
+Objects of delimited composite types that are \emph{not} nested inside other objects (i.e., top-level objects)
+are serialized as if they were of a final type (without the delimiter header).
+The delimiter header, therefore, logically belongs to the container object rather than the contained one.
+
+\begin{remark}
+    Top-level objects do not require the delimiter header because the change in their length does not necessarily
+    affect the backward compatibility thanks to the implicit truncation rule
+    (section \ref{sec:dsdl_serialization_implicit_truncation}) and the implicit zero extension rule
+    (section \ref{sec:dsdl_serialization_implicit_zero_extension}).
+\end{remark}
+
+The delimiter header is an implicit field of type \verb|uint32| that encodes the length of the following
+serialized representation in bytes\footnote{%
```

Comment:
Is this an apt description though? It really just encodes the offset in bytes that the next field starts at, right? The length of the datatype is still a property of the definition the transmitting node used and any variable-length array lengths.

Reply:
> It really just encodes the offset in bytes that the next field starts at, right?

Yes.